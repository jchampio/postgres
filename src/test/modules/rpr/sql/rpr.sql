-- Singleton sets.
\! rpr_prefer a
\! rpr_prefer $
\! rpr_prefer ^
\! rpr_prefer "()"

-- Question mark quantifier (greedy and reluctant) and its verbose equivalent.
\! rpr_prefer a?
\! rpr_prefer a{0,1}
\! rpr_prefer a{,1}
\! rpr_prefer a??
\! rpr_prefer a{0,1}?
\! rpr_prefer a{,1}?

-- Bounded quantifiers.
\! rpr_prefer a{2}
\! rpr_prefer a{2,2}
\! rpr_prefer a{2,2}?
\! rpr_prefer a{1,3}
\! rpr_prefer a{1,3}?
\! rpr_prefer "^{3}"

\! rpr_prefer "(a|b){1,3}"
\! rpr_prefer "(a|b){1,3}?"

-- Empty matches do not appear in the factor's PL except at the very end or in
-- the positions prior to the quantifier's lower bound.
\! rpr_prefer "^{3,10}"
\! rpr_prefer "(a?){1,3}"
\! rpr_prefer "(a?){1,3}?"
\! rpr_prefer "(a?){2,3}"
\! rpr_prefer "(a?){2,3}?"
\! rpr_prefer "(a??){2,3}"
\! rpr_prefer "(a??){2,3}?"

-- Unbounded quantifiers, with a --max-rows setting.
\! rpr_prefer -m 3 "a*"
\! rpr_prefer -m 3 "a{0,}"
\! rpr_prefer -m 3 "a{,}"
\! rpr_prefer -m 3 "a*?"
\! rpr_prefer -m 3 "a{0,}?"
\! rpr_prefer -m 3 "a{,}?"

\! rpr_prefer -m 3 "a+"
\! rpr_prefer -m 3 "a{1,}"
\! rpr_prefer -m 3 "a+?"
\! rpr_prefer -m 3 "a{1,}?"

-- Parenthesization.
\! rpr_prefer "(())"
\! rpr_prefer "(a?)"

-- Exclusion.
\! rpr_prefer "{- a -}"
\! rpr_prefer "{- a b -}"
\! rpr_prefer "a {- a b -} b"

-- Concatenation.
\! rpr_prefer "a b"
\! rpr_prefer "a b?"
\! rpr_prefer "a b c"
\! rpr_prefer "a b c?"
\! rpr_prefer "a b? c"
\! rpr_prefer "a b? c?"
\! rpr_prefer "(a b)"

-- Alternation.
\! rpr_prefer "a|b"
\! rpr_prefer "a?|b?"
\! rpr_prefer "a??|b?"
\! rpr_prefer "a|b|c"
\! rpr_prefer "a|(b|c)"
\! rpr_prefer "a|(b|c)?"
\! rpr_prefer "(a|b)(c|d)"
