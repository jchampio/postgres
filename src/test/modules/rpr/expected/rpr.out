-- Singleton sets.
\! rpr_prefer a
a 
\! rpr_prefer $
$ 
\! rpr_prefer ^
^ 
\! rpr_prefer "()"
( ) 
-- Question mark quantifier (greedy and reluctant) and its verbose equivalent.
\! rpr_prefer a?
( a ) 
( ) 
\! rpr_prefer a{0,1}
( a ) 
( ) 
\! rpr_prefer a{,1}
( a ) 
( ) 
\! rpr_prefer a??
( ) 
( a ) 
\! rpr_prefer a{0,1}?
( ) 
( a ) 
\! rpr_prefer a{,1}?
( ) 
( a ) 
-- Bounded quantifiers.
\! rpr_prefer a{2}
( a a ) 
\! rpr_prefer a{2,2}
( a a ) 
\! rpr_prefer a{2,2}?
( a a ) 
\! rpr_prefer a{1,3}
( a a a ) 
( a a ) 
( a ) 
\! rpr_prefer a{1,3}?
( a ) 
( a a ) 
( a a a ) 
\! rpr_prefer "^{3}"
( ^ ^ ^ ) 
\! rpr_prefer "(a|b){1,3}"
( ( ( a - ) ) ( ( a - ) ) ( ( a - ) ) ) 
( ( ( a - ) ) ( ( a - ) ) ( ( - b ) ) ) 
( ( ( a - ) ) ( ( a - ) ) ) 
( ( ( a - ) ) ( ( - b ) ) ( ( a - ) ) ) 
( ( ( a - ) ) ( ( - b ) ) ( ( - b ) ) ) 
( ( ( a - ) ) ( ( - b ) ) ) 
( ( ( a - ) ) ) 
( ( ( - b ) ) ( ( a - ) ) ( ( a - ) ) ) 
( ( ( - b ) ) ( ( a - ) ) ( ( - b ) ) ) 
( ( ( - b ) ) ( ( a - ) ) ) 
( ( ( - b ) ) ( ( - b ) ) ( ( a - ) ) ) 
( ( ( - b ) ) ( ( - b ) ) ( ( - b ) ) ) 
( ( ( - b ) ) ( ( - b ) ) ) 
( ( ( - b ) ) ) 
\! rpr_prefer "(a|b){1,3}?"
( ( ( a - ) ) ) 
( ( ( a - ) ) ( ( a - ) ) ) 
( ( ( a - ) ) ( ( a - ) ) ( ( a - ) ) ) 
( ( ( a - ) ) ( ( a - ) ) ( ( - b ) ) ) 
( ( ( a - ) ) ( ( - b ) ) ) 
( ( ( a - ) ) ( ( - b ) ) ( ( a - ) ) ) 
( ( ( a - ) ) ( ( - b ) ) ( ( - b ) ) ) 
( ( ( - b ) ) ) 
( ( ( - b ) ) ( ( a - ) ) ) 
( ( ( - b ) ) ( ( a - ) ) ( ( a - ) ) ) 
( ( ( - b ) ) ( ( a - ) ) ( ( - b ) ) ) 
( ( ( - b ) ) ( ( - b ) ) ) 
( ( ( - b ) ) ( ( - b ) ) ( ( a - ) ) ) 
( ( ( - b ) ) ( ( - b ) ) ( ( - b ) ) ) 
-- Empty matches do not appear in the factor's PL except at the very end or in
-- the positions prior to the quantifier's lower bound.
\! rpr_prefer "^{3,10}"
( ^ ^ ^ ) 
\! rpr_prefer "(a?){1,3}"
( ( ( a ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ) 
( ( ( ) ) ) 
\! rpr_prefer "(a?){1,3}?"
( ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ( ( ) ) ) 
( ( ( ) ) ) 
\! rpr_prefer "(a?){2,3}"
( ( ( a ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( ) ) ) 
( ( ( ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( ) ) ( ( a ) ) ) 
( ( ( ) ) ( ( ) ) ) 
\! rpr_prefer "(a?){2,3}?"
( ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ( ( ) ) ) 
( ( ( ) ) ( ( a ) ) ) 
( ( ( ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( ) ) ( ( ) ) ) 
\! rpr_prefer "(a??){2,3}"
( ( ( ) ) ( ( ) ) ) 
( ( ( ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ) 
\! rpr_prefer "(a??){2,3}?"
( ( ( ) ) ( ( ) ) ) 
( ( ( ) ) ( ( a ) ) ) 
( ( ( ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( ) ) ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ( ( a ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( ) ) ) 
( ( ( a ) ) ( ( a ) ) ( ( a ) ) ) 
-- Unbounded quantifiers, with a --max-rows setting.
\! rpr_prefer -m 3 "a*"
( a a a ) 
( a a ) 
( a ) 
( ) 
\! rpr_prefer -m 3 "a{0,}"
( a a a ) 
( a a ) 
( a ) 
( ) 
\! rpr_prefer -m 3 "a{,}"
( a a a ) 
( a a ) 
( a ) 
( ) 
\! rpr_prefer -m 3 "a*?"
( ) 
( a ) 
( a a ) 
( a a a ) 
\! rpr_prefer -m 3 "a{0,}?"
( ) 
( a ) 
( a a ) 
( a a a ) 
\! rpr_prefer -m 3 "a{,}?"
( ) 
( a ) 
( a a ) 
( a a a ) 
\! rpr_prefer -m 3 "^*"
( ^ ) 
( ) 
\! rpr_prefer -m 3 "^*?"
( ) 
( ^ ) 
\! rpr_prefer -m 3 "a+"
( a a a ) 
( a a ) 
( a ) 
\! rpr_prefer -m 3 "a{1,}"
( a a a ) 
( a a ) 
( a ) 
\! rpr_prefer -m 3 "a+?"
( a ) 
( a a ) 
( a a a ) 
\! rpr_prefer -m 3 "a{1,}?"
( a ) 
( a a ) 
( a a a ) 
\! rpr_prefer -m 3 "^+"
( ^ ) 
-- Empty match rules must still apply.
\! rpr_prefer -m 3 "(){0,}"
( ( ) ) 
( ) 
\! rpr_prefer -m 3 "(){1,}"
( ( ) ) 
\! rpr_prefer -m 3 "(){2,}"
( ( ) ( ) ) 
-- Parenthesization.
\! rpr_prefer "(())"
( ( ) ) 
\! rpr_prefer "(a?)"
( ( a ) ) 
( ( ) ) 
-- Exclusion.
\! rpr_prefer "{- a -}"
[ a ] 
\! rpr_prefer "{- a b -}"
[ ( a b ) ] 
\! rpr_prefer "a {- a b -} b"
( ( a [ ( a b ) ] ) b ) 
-- Concatenation.
\! rpr_prefer "a b"
( a b ) 
\! rpr_prefer "a b?"
( a ( b ) ) 
( a ( ) ) 
\! rpr_prefer "a b c"
( ( a b ) c ) 
\! rpr_prefer "a b c?"
( ( a b ) ( c ) ) 
( ( a b ) ( ) ) 
\! rpr_prefer "a b? c"
( ( a ( b ) ) c ) 
( ( a ( ) ) c ) 
\! rpr_prefer "a b? c?"
( ( a ( b ) ) ( c ) ) 
( ( a ( b ) ) ( ) ) 
( ( a ( ) ) ( c ) ) 
( ( a ( ) ) ( ) ) 
\! rpr_prefer "(a b)"
( ( a b ) ) 
-- Alternation.
\! rpr_prefer "a|b"
( a - ) 
( - b ) 
\! rpr_prefer "a?|b?"
( ( a ) - ) 
( ( ) - ) 
( - ( b ) ) 
( - ( ) ) 
\! rpr_prefer "a??|b?"
( ( ) - ) 
( ( a ) - ) 
( - ( b ) ) 
( - ( ) ) 
\! rpr_prefer "a|b|c"
( ( a - ) - ) 
( ( - b ) - ) 
( - c ) 
\! rpr_prefer "a|(b|c)"
( a - ) 
( - ( ( b - ) ) ) 
( - ( ( - c ) ) ) 
\! rpr_prefer "a|(b|c)?"
( a - ) 
( - ( ( ( b - ) ) ) ) 
( - ( ( ( - c ) ) ) ) 
( - ( ) ) 
\! rpr_prefer "(a|b)(c|d)"
( ( ( a - ) ) ( ( c - ) ) ) 
( ( ( a - ) ) ( ( - d ) ) ) 
( ( ( - b ) ) ( ( c - ) ) ) 
( ( ( - b ) ) ( ( - d ) ) ) 
